lib/ykpiv.c

```diff
@@ -704,6 +761,11 @@ ykpiv_rc ykpiv_verify(ykpiv_state *state, const char *pin, int *tries) {
   if((res = send_data(state, &apdu, data, &recv_len, &sw)) != YKPIV_OK) {
     return res;
   } else if(sw == SW_SUCCESS) {
+    if (pin) {
+      free(state->pin);
+      state->pin = malloc(len * sizeof(char) + 1);
```

Comment:
It would be nice to check that malloc() succeeds here, return YKPIV_MEMORY_ERROR, or maybe it should only allocate when the state allocates to avoid another error-case here..